### ISSUE TITLE
Options are not being applied to prompt during generation

### ISSUE THREAD

**Issue by ezynda3** [teilomillet/gollm#13]
Given the following example
```go
package main

import (
	"context"
	"fmt"
	"log"
	"os"
	"time"

	"github.com/teilomillet/gollm"
)

func main() {
	// Load API key from environment variable
	apiKey := os.Getenv("OPENAI_API_KEY")
	if apiKey == "" {
		log.Fatalf("OPENAI_API_KEY environment variable is not set")
	}

	// Create a new LLM instance with custom configuration
	llm, err := gollm.NewLLM(
		gollm.SetProvider("openai"),
		gollm.SetModel("gpt-4o-mini"),
		gollm.SetAPIKey(apiKey),
		gollm.SetMaxTokens(300),
		gollm.SetMaxRetries(3),
		gollm.SetRetryDelay(time.Second*2),
		gollm.SetLogLevel(gollm.LogLevelDebug),
	)
	if err != nil {
		log.Fatalf("Failed to create LLM: %v", err)
	}

	ctx := context.Background()
	prompt := gollm.NewPrompt("What is a florb?",
		gollm.WithContext("A florb is a red cube"),
		gollm.WithDirectives("Be concise", "Use examples"),
		gollm.WithOutput("Output in JSON format"),
		gollm.WithMaxLength(300),
	)
	response, err := llm.Generate(ctx, prompt)
	fmt.Println(response)
}
```

I get the following debug log
```
2024/11/04 14:54:23 DEBUG: Starting Generate method [prompt_length 226 context context.Background]
2024/11/04 14:54:23 DEBUG: Generating text [provider openai prompt What is a florb? system_prompt  attempt 1]
2024/11/04 14:54:23 DEBUG: Full request body [body {"max_tokens":300,"messages":[{"content":"What is a florb?","role":"user"}],"model":"gpt-4o-mini","temperature":0.7}]
2024/11/04 14:54:23 DEBUG: Full API request [method POST url https://api.openai.com/v1/chat/completions headers map[] body {"max_tokens":300,"messages":[{"content":"What is a florb?","role":"user"}],"model":"gpt-4o-mini","temperature":0.7}]
```

It appears that only `Prompt.Input` is actually being sent to the provider for generation while the rest of the options are ignored...

**Comment by ezynda3** [comment]
Looks like the bug was probably introduced in this commit.

https://github.com/teilomillet/gollm/commit/1da69896c20755860170806fba188ad48aea2fef#diff-f4b356a7b15ee425318c5d670a1cd20a6f91441a484282a10e0cf1a68b1bd94aL95-R114

**Comment by ezynda3** [comment]
Submitted a PR https://github.com/teilomillet/gollm/pull/14

**Comment by Dabnis** [comment]
This is why we have unit testing! Pre merge

**Comment by ezynda3** [comment]
Fixed with #14
